hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettag/TargetTagFilterButtons.java

```diff
@@ -213,35 +210,31 @@ private boolean checkForTargetUpdatePermission() {
     }
 
     private void processTargetDrop(final DragAndDropEvent event) {
-
         final com.vaadin.event.dd.TargetDetails targetDetails = event.getTargetDetails();
         final TableTransferable transferable = (TableTransferable) event.getTransferable();
-        final Table source = transferable.getSourceComponent();
-
-        final Set<TargetIdName> targetSelected = AbstractTable.getTableValue(source);
-        final Set<String> targetList = new HashSet<>();
-        if (transferable.getData(ITEMID) != null) {
-            if (!targetSelected.contains(transferable.getData(ITEMID))) {
-                targetList.add(((TargetIdName) transferable.getData(ITEMID)).getControllerId());
-            } else {
-                targetList.addAll(targetSelected.stream().map(t -> t.getControllerId()).collect(Collectors.toList()));
-            }
+        @SuppressWarnings("unchecked")
+        final AbstractTable<?, TargetIdName> targetTable = (AbstractTable<?, TargetIdName>) transferable
+                .getSourceComponent();
 
-            final String targTagName = HawkbitCommonUtil.removePrefix(targetDetails.getTarget().getId(),
-                    SPUIDefinitions.TARGET_TAG_ID_PREFIXS);
+        final Set<TargetIdName> targetSelected = targetTable.getDeletedEntityByTransferable(transferable);
+        final Set<String> targetList = targetSelected.stream().map(t -> t.getControllerId())
+                .collect(Collectors.toSet());
 
-            final List<String> tagsClickedList = managementUIState.getTargetTableFilters().getClickedTargetTags();
+        final String targTagName = HawkbitCommonUtil.removePrefix(targetDetails.getTarget().getId(),
+                SPUIDefinitions.TARGET_TAG_ID_PREFIXS);
 
-            final TargetTagAssignmentResult result = targetManagement.toggleTagAssignment(targetList, targTagName);
-            notification.displaySuccess(HawkbitCommonUtil.createAssignmentMessage(targTagName, result, i18n));
+        final List<String> tagsClickedList = managementUIState.getTargetTableFilters().getClickedTargetTags();
 
-            if (result.getAssigned() >= 1 && managementUIState.getTargetTableFilters().isNoTagSelected()) {
-                eventBus.publish(this, ManagementUIEvent.ASSIGN_TARGET_TAG);
-            }
-            if (result.getUnassigned() >= 1 && !tagsClickedList.isEmpty() && tagsClickedList.contains(targTagName)) {
-                eventBus.publish(this, ManagementUIEvent.UNASSIGN_TARGET_TAG);
-            }
+        final TargetTagAssignmentResult result = targetManagement.toggleTagAssignment(targetList, targTagName);
+        notification.displaySuccess(HawkbitCommonUtil.createAssignmentMessage(targTagName, result, i18n));
+
+        if (result.getAssigned() >= 1 && managementUIState.getTargetTableFilters().isNoTagSelected()) {
```

Comment:
Extract conditions to new private method. So you can express the comment with a proper method name